agent/xds/listeners.go

```diff
@@ -289,7 +289,50 @@ func (s *ResourceGenerator) listenersFromSnapshotConnectProxy(cfgSnap *proxycfg.
 		// Below we create a filter chain per upstream, rather than a listener per upstream
 		// as we do for explicit upstreams above.
 
-		// TODO(peering): tproxy
+		filterChain, err := s.makeUpstreamFilterChain(filterChainOpts{
+			routeName:   uid.EnvoyID(),
+			clusterName: clusterName,
+			filterName:  uid.EnvoyID(),
```

Comment:
this diverges from disco chain filter names but is the same name format we give explicit peered upstreams. Thoughts?
```go
filterName := fmt.Sprintf("%s.%s.%s.%s", chain.ServiceName, chain.Namespace, chain.Partition, chain.Datacenter)
```

Reply:
Seems fine to me and I think having the query param style syntax that EnvoyID uses will make it easier to understand/debug if you are looking at a config dump.

Reply:
Should we update the disco chain filter names (in the future) to use this syntax?